lib/supavisor.ex

```diff
@@ -40,19 +40,20 @@ defmodule Supavisor do
     end
   end
 
-  @spec subscribe_local(pid, String.t(), String.t()) :: {:ok, workers} | {:error, any()}
+  @spec subscribe_local(pid, String.t(), String.t()) :: {:ok, workers, iodata()} | {:error, any()}
   def subscribe_local(pid, tenant, user_alias) do
     with {:ok, workers} <- get_local_workers(tenant, user_alias),
-         :ok <- Manager.subscribe(workers.manager, pid) do
-      {:ok, workers}
+         {:ok, ps} <- Manager.subscribe(workers.manager, pid) do
+      Process.monitor(workers.manager)
```

Comment:
Where is the `:DOWN` for this monitor handled?

Reply:
in the client connection https://github.com/supabase/supavisor/blob/68979bb6448e38bf5efe8daf502399358bd61e62/lib/supavisor/client_handler.ex#L195-L207

Reply:
So maybe it is preferrable to monitor it there? So we monitor it close to where the DOWN is received?

Reply:
Yep, that would be better, thanks for finding that!